core/src/services/s3/backend.rs

```diff
@@ -1311,6 +1348,7 @@ impl S3Backend {
         content_type: Option<&str>,
         content_disposition: Option<&str>,
         cache_control: Option<&str>,
+        storage_class: Option<&str>,
```

Comment:
looks like the `s3_put_object_request` method already has many arguments 🤣

Reply:
https://github.com/apache/incubator-opendal/actions/runs/4616153426/jobs/8160810398?pr=1854 ooops clippy is unhappy about this method, should we refactor it a bit?

Reply:
> should we refactor it a bit?

Yes, but it's Ok to keep moving on by add some `allow`. I prefer to refactor in another PR.

Reply:
> I prefer to refactor in another PR.

I might be able to do that. Which direction should we take? Adding a config struct comprised of `content_type`, `content_disposition`, `cache_control` and `storage_class`? as they are all headers which could be set in the write request.

Reply:
I think we can pass a `OpWrite` directly.

Reply:
It turns out that we don't need to modify the signature of this method now